t/shared.t

```diff
@@ -33,5 +32,23 @@ is( $hash{multi}->{key1},      "foo", "multidimension, key1" );
 is( $hash{multi}->{arr1}->[0], "bar", "multidimension, arr1 - key0" );
 is( $hash{multi}->{arr1}->[1], "baz", "multidimension, arr1 - key1" );
 
-unlink("vars.db");
-unlink("vars.db.lock");
+{
+    @array = (0);
+    my @pids;
+    for my $i (0..5) {
+        my $pid = fork();
+        if ($pid == 0) {
+            # child
+            sleep 1;
```

Comment:
Well, I have a strong feeling against hardcoded `sleep`s, so I'm trying hard to find an alternative without it :)

I think we can wait "just enough" time if we poll all the childs in a while loop with SIGZERO, and if they all return success, send SIGCONT. Plus there can be an `alarm` for 2 seconds, to abort the test with failure, instead of hanging.

I'm willing to spend more effort on this, but obviously I could use a second pair of eyes, and extra testing, as it seems we already have different results for the same code. I ran it 100 times both without and with your patches, and I could reproducibly make it fail or pass.